this shit is wrong, but so was original ... this needs to be
rewritten so that $HOST_ARCH/$TARGET_ARCH aren't even needed ...

--- configure
+++ configure
@@ -129,8 +129,13 @@
 #
 # 0. Figure out architecture (one of i386, alpha, sparc, arm, m68k, mips)
 #
-ARCH=`uname -m | sed s/i.86/i386/`
-echo "ARCH=$ARCH" >> make_include
+HOST_ARCH=`uname -m`
+TARGET_ARCH=`$CC -dumpmachine | awk -F- '{print $1}'`
+# Don't allow ARCH="sparc" for sparc64 targets with 32-bit userland
+if (test "$HOST_ARCH" = "sparc64" && test "$TARGET_ARCH" = "sparc") ; then
+	TARGET_ARCH="sparc64"
+fi
+echo "ARCH=${TARGET_ARCH}" | sed -e 's:i.86:i386:' >> make_include
 
 #
 # 1. Do we have <locale.h>?
